### PR TITLE
Fix INSERT with only default values

### PR DESCRIPTION
**Pull request by stanislas-m**
INSERT using default values for all columns made a syntax error on most databases.
Also split migrations test patches for better management, removed unused field in belongs_to association and removed the old zero value check function.

### CODE REVIEW

File: associations/belongs_to_association.go [hunk 1]
```diff
@@ -70,7 +69,6 @@ func belongsToAssociationBuilder(p associationParams) (Association, error) {
 		ownerModel: fval,
 		ownerType:  fval.Type(),
 		ownerID:    f,
-		fkID:       ownerIDField,
```
Comment (mclark4386) [inline associations/belongs_to_association.go]: I feel like this may be important when we start working eager/flat update.

Comment (stanislas-m) [reply associations/belongs_to_association.go]: If it's needed in the future, we'll add it in the future. :)
